crates/telio-wg/src/wg.rs

```diff
@@ -470,30 +477,36 @@ impl State {
     }
 
     fn time_since_last_rx(&self, public_key: PublicKey) -> Option<Duration> {
-        self.last_rx_timestamp
+        self.additional_peer_info
             .get(&public_key)
-            .map(|v| Instant::now() - v.0)
+            .map(|p| Instant::now() - p.last_rx_timestamp)
     }
 
     #[allow(mpsc_blocking_send)]
     async fn update_send_notification_events(
         &self,
         to: &uapi::Interface,
         diff_keys: &DiffKeys,
+        new_additional_peer_info: &HashMap<PublicKey, AdditionalPeerInfo>,
     ) -> bool {
         let from = &self.interface;
 
         // Setup
-        let try_send = |state, peer| {
+        let try_send = |state, link_state, peer| {
             self.event
-                .send(Box::new(Event { state, peer }))
+                .send(Box::new(Event {
+                    state,
+                    link_state,
+                    peer,
+                }))
                 .map(|r| r.is_ok())
         };
 
         // Notify all disconnects
         for key in &diff_keys.delete_keys {
             if !try_send(
                 PeerState::Disconnected,
+                LinkState::Down,
```

Comment:
I think we should have an `Option` with `None` in this case as we should only care about the `LinkState` when we're connected